src/Features/CSharp/Portable/Completion/SuggestionMode/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -41,6 +41,10 @@ internal class CSharpSuggestionModeCompletionProvider : SuggestionModeCompletion
                 {
                     return CreateSuggestionModeItem(CSharpFeaturesResources.lambda_expression, CSharpFeaturesResources.Autoselect_disabled_due_to_potential_lambda_declaration);
                 }
+                if (IsRangeExpression(semanticModel, token, cancellationToken))
+                {
+                    return CreateSuggestionModeItem(CSharpFeaturesResources.range_expression, CSharpFeaturesResources.Autoselect_disabled_due_to_potential_range_expression);
```

Comment:
i don't think this is waht we want.  instead, i think we just want to change how completin works such that *if* we think we're in a range-context (which we shuld use similar logic to doing 'lambda contxt' detection), then we change how commit works so that `.` *only* commits if you've typed any characters.  That way if you're in a range context and you type `foo<dot><dot>` you get `foo..`.  but if you type `foo<dot>b<dot>` you end up with `foo.bar.`

Reply:
Thank you, @cyrusnajmabadi! Agree.

Will investigate other options rather than suggestion mode for this.

I'm not sure I understand the idea for the lambda context. Do you mean we should correct it as well? Or do you mean it is implemented properly and we should borrow the approach?

---
In reply to: [284524359](https://github.com/dotnet/roslyn/pull/35749#discussion_r284524359) [](ancestors = 284524359)